CryptoLib/CryptoLib/Ldap/OpenLdap.swift

```diff
@@ -101,13 +101,21 @@ public class OpenLdap {
             .replacingOccurrences(of: ")", with: "\\)")
             .replacingOccurrences(of: "*", with: "\\*")
 
-        let filter = if isPersonalCode(escapedIdentityCode) {
-            "(serialNumber=\(secureLdap ? "PNOEE-" : "")\(escapedIdentityCode))"
-        } else if escapedIdentityCode.count > 4 && escapedIdentityCode.rangeOfCharacter(from: CharacterSet.decimalDigits.inverted) == nil {
-            "(serialNumber=\(escapedIdentityCode))"
+        let isDigitOnly = escapedIdentityCode.rangeOfCharacter(from: CharacterSet.decimalDigits.inverted) == nil
+        let length = escapedIdentityCode.count
+
+        let filter: String
+
+        if !isDigitOnly || (length != 8 && length != 11) {
+            filter = "(cn=*\(escapedIdentityCode)*)"
+        } else if length == 8 {
+            filter = "(serialNumber=\(escapedIdentityCode))"
+        } else if length == 11 && isPersonalCode(escapedIdentityCode) {
+            filter = "(serialNumber=\(secureLdap ? "PNOEE-" : "")\(escapedIdentityCode))"
```

Comment:
```suggestion
var filter: String = switch (isDigitOnly, length) {
case (true, 8): "(serialNumber=\(escapedIdentityCode))"
case (true, 11): "(serialNumber=PNOEE-\(escapedIdentityCode))"
default: "(cn=*\(escapedIdentityCode)*)"
}
```
this is more readable and isPersonalCode can be removed because there is no more magic there